From c5eaead982b091983d80d97a819894bd2e7b2cbf Mon Sep 17 00:00:00 2001
From: Pavithra Ganesan <quic_pavigane@quicinc.com>
Date: Wed, 21 Aug 2024 17:26:59 +0530
Subject: [PATCH] [Revert] [UPSTREAM] hostapd: Set netdev mac addr as MLD addr

Fix the different bssid and mld addr in SLO.

Addresses being assigned to the interface which changes across
reboot or interface deletion and creation, so far we might
not encountered this since hostapd sets the bssid to
the iface every time on bss init(now this will not
happen to have netdev mac addr as mld addr).

Reverting below upstream commit

commit f60287e6c91bbb5f20758a6f5000ec9fd714fae2
Author: Andrei Otcheretianski <andrei.otcheretianski@intel.com>
Date:   Tue Nov 21 01:51:39 2023 +0200

    AP: Avoid setting same MLD and link address

    For AP MLD, when BSSID configuration is specified without mld_addr, the
    first link address is used as the MLD MAC address as well. Though IEEE
    P802.11be allows this, the current implementation and the kernel aren't
    happy about it. Better avoid this.

Signed-off-by: Pavithra Ganesan <quic_pavigane@quicinc.com>
---
 hostapd/main.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

--- a/hostapd/main.c
+++ b/hostapd/main.c
@@ -230,12 +230,8 @@ static int hostapd_driver_init(struct ho
 	 * Use the configured MLD MAC address as the interface hardware address
 	 * if this AP is a part of an AP MLD.
 	 */
-	if (hapd->conf->mld_ap) {
-		if (!is_zero_ether_addr(hapd->conf->mld_addr))
-			params.bssid = hapd->conf->mld_addr;
-		else
-			params.bssid = NULL;
-	}
+	if (!is_zero_ether_addr(hapd->conf->mld_addr) && hapd->conf->mld_ap)
+		params.bssid = hapd->conf->mld_addr;
 #endif /* CONFIG_IEEE80211BE */
 
 	params.ifname = hapd->conf->iface;
